### PR TITLE
Remove old checkpoint dir if it still exists

### LINKED ISSUE #6619: When checkpoint is not successful, next checkpoint is twice the size.
## Bug Report
**What did you do?**

Start prometheus with aggressive settings (--storage.tsdb.max-block-duration=10m --storage.tsdb.min-block-duration=1m)

Wait for a checkpoint:
data/wal/checkpoint.000040/

Let's say that the checkpoint has a size of 5M

Copy the checkpoint to data/wal/checkpoint.000042.tmp

_It simulates the case where the tmp checkpoint is not written completely_

**What did you expect to see?**

when checkpoint 0000042 is written, its size is 5M

**What did you see instead? Under which circumstances?**

checkpoint 0000042 is 10M, because it contains the tmp that was not written successfully.

**Environment**

Linux

* Prometheus version:

2.15.2

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

Fix #6619

### CODE REVIEW

File: tsdb/wal/checkpoint.go [hunk 1]
```diff
@@ -133,6 +133,13 @@ func Checkpoint(w *WAL, from, to int, keep func(id uint64) bool, mint int64) (*C
 	cpdir := filepath.Join(w.Dir(), fmt.Sprintf(checkpointPrefix+"%06d", to))
 	cpdirtmp := cpdir + ".tmp"
 
+	if _, err := os.Stat(cpdirtmp); !os.IsNotExist(err) {
```
Comment (brian-brazil) [inline tsdb/wal/checkpoint.go]: This error check feels off. How about always trying, and then use the error check on the removal?

Comment (roidelapluie) [reply tsdb/wal/checkpoint.go]: ack

Comment (roidelapluie) [reply tsdb/wal/checkpoint.go]: done

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**